nipy/algorithms/statistics/formula/formulae.py

```diff
@@ -575,33 +585,47 @@ def __add__(self, other):
         """
         if not is_formula(other):
             raise ValueError('only Formula objects can be added to a Formula')
-        f = self.__class__(np.hstack([self.terms, other.terms]))
+        f = Formula(np.hstack([self.terms, other.terms]))
         return f
 
     def __sub__(self, other):
-        """
-        Create a new Formula by deleting terms in other
-        from self. No exceptions are raised for terms in other that do not appear in
-        self.
+        """ Formula from deleting terms in `other` from those in `self`
+
+        No exceptions are raised for terms in `other` that do not appear in
+        `self`.
+
+        Parameters
+        ----------
+        other : Formula instance
+            Object for which ``is_formula(other)`` is True
+
+        Returns
+        -------
+        subbed : Formula instance
+            Formula with terms of `other` removed from terms of `self`
 
+        Examples
+        --------
         >>> x, y, z = [Term(l) for l in 'xyz']
-        >>> f1 = Formula([x,y,z])
-        >>> f2 = Formula([y])+I
+        >>> f1 = Formula([x, y, z])
+        >>> f2 = Formula([y]) + I
         >>> f1.mean
         _b0*x + _b1*y + _b2*z
         >>> f2.mean
         _b0*y + _b1
-        >>> f3=f2-f1
+        >>> f3 = f2 - f1
         >>> f3.mean
         _b0
-        >>> f4=f1-f2
+        >>> f4 = f1 - f2
         >>> f4.mean
         _b0*x + _b1*z
         """
         if not is_formula(other):
-            raise ValueError('only Formula objects can be subtracted from a Formula')
-        d = list(set(self.terms).difference(other.terms))
-        return self.__class__(d)
+            raise ValueError(
+                'only Formula objects can be subtracted from a Formula')
+        # Preserve order of terms in subtraction
+        d = [term for term in self.terms if term not in other.terms]
+        return Formula(d)
```

Comment:
there was a time when python folks promised the death of filter superseeded by list comprehensions.... everyone shifted but I find myself using filter/map more since they are easier to read
```python
d = filter(lambda term: term not in other.terms, self.terms)
```
the only thing is that this one is not python3-compatible so if you use six you would need to
```
from six.moves import filter
d = list(filter(lambda term: term not in other.terms, self.terms))
```